great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1020,14 +1011,10 @@ def _serialize_substitute_and_sanitize_datasource_config(
         Returns:
             Dict of config with substitutions and sanitizations applied.
         """
-        substitutions: dict = self._determine_substitutions()
         datasource_dict: dict = serializer.serialize(datasource_config)
 
         substituted_config: dict = cast(
-            dict,
-            substitute_all_config_variables(
-                datasource_dict, substitutions, self.DOLLAR_SIGN_ESCAPE_STRING
-            ),
+            dict, self.config_provider.substitute_config(datasource_dict)
```

Comment:
You could removed the `: dict` here since you are explicitly casting it.

Reply:
done!